cvmfs/cache_ram.cc

```diff
@@ -0,0 +1,250 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+#include "cvmfs_config.h"
+#include "cache_ram.h"
+
+#include <errno.h>
+#include <assert.h>
+#include <new>
+#include <string.h>
+
+#include "kvstore.h"
+#include "smalloc.h"
+#include "util/posix.h"
+#include "util_concurrency.h"
+
+using namespace std;  // NOLINT
+
+namespace cache {
+
+
+int RamCacheManager::AddFd(const ReadOnlyFd &fd) {
```

Comment:
Further, you probably want to keep a "free list" of known invalid handles to avoid a linear scan of the data structure when opening / closing files in the cache.

Something like this:

```
  if (free_fd_.empty()) {  // No holes in map; add new int.
    i = open_fds_.size();
    open_fds_.push_back(fd);
  } else {
    i = free_fd_.back();
    free_fd_.pop_back();
    open_fd_[i] = fd;
  }
  return i;
```